geopandas/tests/test_geoseries.py

```diff
@@ -159,6 +159,24 @@ def test_geom_equals_exact(self):
             self.a1.geom_equals_exact(self.a2, 0.001, align=False), [False, False]
         )
 
+    @pytest.mark.skipif(not compat.SHAPELY_GE_21, reason="requires Shapely>=2.1")
+    def test_geom_equals_identical(self):
+        assert np.all(self.g1.geom_equals_identical(self.g1))
+        assert_array_equal(self.g1.geom_equals_identical(self.sq), [False, True])
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                "The indices of the left and right GeoSeries' are not equal",
+                UserWarning,
+            )
```

Comment:
this filterwarnings is redundant I think, it would only be needed if align=None?